enterprise/server/remote_execution/containers/firecracker/firecracker.go

```diff
@@ -176,6 +176,34 @@ func openFile(ctx context.Context, env environment.Env, fileName string) (io.Rea
 	return env.GetFileResolver().Open(fileName)
 }
 
+// Returns the cgroup version available on this machine, which is returned from
+//
+//	stat -fc %T /sys/fs/cgroup/
+//
+// and should be either 'cgroup2fs' (version 2) or 'tmpfs' (version 1)
+//
+// More info here: https://kubernetes.io/docs/concepts/architecture/cgroups/
+// TODO(iain): preemptively get this in a provider a la Provider in podman.go.
+func getCgroupVersion() (string, error) {
+	if *firecrackerCgroupVersion != "" {
+		return *firecrackerCgroupVersion, nil
+	}
+	b, err := exec.Command("stat", "-fc", "%T", "/sys/fs/cgroup/").Output()
+	if err != nil {
+		log.Debugf("Error determining system cgroup version: %s", err.Error())
```

Comment:
the caller is likely already going to report/log the error so instead of a separate log statement it would be better to enrich the returned error message with more information, something like

`return "", status.UnavailableErrorf("error determining...: %s", err)`